core/src/main/java/org/apache/accumulo/core/client/summary/CountingSummarizer.java

```diff
@@ -0,0 +1,302 @@
+ *     }
+ *
+ *     &#064;Override
+ *     protected Converter&lt;ByteSequence&gt; converter() {
+ *       return (key, val, consumer) -&gt; consumer.accept(key.getColumnVisibilityData());
+ *     }
+ *   }
+ * </code>
+ * </pre>
+ *
+ * @param <K>
+ *          The counter key type. This type must have good implementations of {@link Object#hashCode()} and {@link Object#equals(Object)}.
+ * @see CounterSummary
+ * @since 2.0.0
+ */
+//@formatter:on
+public abstract class CountingSummarizer<K> implements Summarizer {
+
+  /**
+   * A configuration option for specifying the maximum number of unique counters an instance of this summarizer should track. If not specified, a default of
+   * {@value #MAX_COUNTER_DEFAULT} will be used.
+   */
+  public static final String MAX_COUNTERS_OPT = "maxCounters";
+
+  /**
+   * A configuration option for specifying the maximum length of an individual counter key. If not specified, a default of {@value #MAX_CKL_DEFAULT} will be
+   * used.
+   */
+  public static final String MAX_COUNTER_LEN_OPT = "maxCounterLen";
+
+  /**
+   * A configuration option to determine if delete keys should be counted. If set to true then delete keys will not be passed to the {@link Converter} and the
+   * statistic {@value #DELETES_IGNORED_STAT} will track the number of deleted ignored. This options defaults to {@value #INGNORE_DELETES_DEFAULT}.
+   */
+  public static final String INGNORE_DELETES_OPT = "ignoreDeletes";
+
+  /**
+   * This prefixes all counters when emitting statistics in {@link Summarizer.Collector#summarize(Summarizer.StatisticConsumer)}.
+   */
+  public static final String COUNTER_STAT_PREFIX = "c:";
+
+  /**
+   * This is the name of the statistic that tracks how many counters objects were ignored because the number of unique counters was exceeded. The max number of
+   * unique counters is specified by {@link #MAX_COUNTERS_OPT}.
+   */
+  public static final String TOO_MANY_STAT = "tooMany";
+
+  /**
+   * This is the name of the statistic that tracks how many counter objects were ignored because they were too long. The maximum lenght is specified by
+   * {@link #MAX_COUNTER_LEN_OPT}.
+   */
+  public static final String TOO_LONG_STAT = "tooLong";
+
+  /**
+   * This is the name of the statistic that tracks the total number of counter objects emitted by the {@link Converter}. This includes emitted Counter objects
+   * that were ignored.
+   */
+  public static final String EMITTED_STAT = "emitted";
+
+  /**
+   * This is the name of the statistic that tracks the total number of deleted keys seen. This statistic is only incremented when the
+   * {@value #INGNORE_DELETES_OPT} option is set to true.
+   */
+  public static final String DELETES_IGNORED_STAT = "deletesIgnored";
+
+  /**
+   * This tracks the total number of key/values seen by the {@link Summarizer.Collector}
+   */
+  public static final String SEEN_STAT = "seen";
+
+  // this default can not be changed as persisted summary data depends on it
```

Comment:
I should add some documentation about considerations for persistence to the super type.